src/main/java/io/debezium/connector/yugabytedb/YugabyteDBTablePoller.java

```diff
@@ -87,69 +92,142 @@ public void run() {
-    try {
-      boolean shouldRestart = false;
-      GetDBStreamInfoResponse resp = this.ybClient.getDBStreamInfo(this.connectorConfig.streamId());
+    // Reset the retry counter.
+    short retryCount = 0;
+    while (retryCount <= MAX_RETRY_COUNT) {
+      try {
+        boolean shouldRestart = false;
+        GetDBStreamInfoResponse resp = this.ybClient.getDBStreamInfo(this.connectorConfig.streamId());
+
+        if (cachedTableInfoSet == null) {
+          LOGGER.debug("Cached table list in the poller thread is null, initializing it now");
+          cachedTableInfoSet = resp.getTableInfoList().stream().collect(Collectors.toSet());
+        } else {
+          if (cachedTableInfoSet.size() != resp.getTableInfoList().size()) {
+            Set<TableInfo> tableInfoSetFromResponse =
+              resp.getTableInfoList().stream().collect(Collectors.toSet());
+
+            Set<TableInfo> cachedSet = new HashSet<>(cachedTableInfoSet);
+            Set<TableInfo> responseSet = new HashSet<>(tableInfoSetFromResponse);
+
+            Set<TableInfo> intersection = new HashSet<>(cachedSet);
+            intersection.retainAll(responseSet);
+
+            Set<TableInfo> difference = new HashSet<>(responseSet);
+            difference.removeAll(cachedSet);
+            
+            if (LOGGER.isDebugEnabled()) {
+              LOGGER.debug("Common tables between the cached table info set and the set received "
+                          + "from GetDBStreamInfoResponse:");
+              intersection.forEach(tableInfo -> {
+                LOGGER.debug(tableInfo.getTableId().toStringUtf8());
+              });
+
+              LOGGER.debug("New tables as received in the GetDBStreamInfoResponse: ");
+              difference.forEach(tableInfo -> {
+                LOGGER.debug(tableInfo.getTableId().toStringUtf8());
+              });
+            }
 
-      // Reset the retry counter.
-      retryCount = 0;
+            for (TableInfo tableInfo : difference) {
+              if (isTableIncludedForStreaming(tableInfo.getTableId().toStringUtf8())) {
+                String message = "Found {} new table(s), signalling context reconfiguration";
+                LOGGER.info(message, difference.size());
+                shouldRestart = true;
+              }
+            }
+          
+            // Update the cached table list.
+            cachedTableInfoSet = tableInfoSetFromResponse;
+          }
+        }
 
-      if (cachedTableInfoSet == null) {
-        LOGGER.debug("Cached table list in the poller thread is null, initializing it now");
-        cachedTableInfoSet = resp.getTableInfoList().stream().collect(Collectors.toSet());
-      } else {
-        if (cachedTableInfoSet.size() != resp.getTableInfoList().size()) {
-          Set<TableInfo> tableInfoSetFromResponse =
-            resp.getTableInfoList().stream().collect(Collectors.toSet());
+        return shouldRestart;
+      } catch (Exception e) {
+        ++retryCount;
```

Comment:
I do not understand the above change, might be because I am missing some info here. Can you please explain a little?

Reply:
In the current implementation, in `areThereNewTablesInStream()` method, we initialize the retry count to zero at entry. Suppose we encounter any error inside the `try` block, we would move to the catch block, increment the retry count and return `false`.  When the method is again invoked after the polling interval, the retry count would be again set to zero. Hence the retries for the failure that was encountered initially, never took place.

In case of publications, we are polling the `pg_publication_tables` to see if new tables were added to the publication, instead of polling over `stream_info`